lib/libc/minimal/include/assert.h

```diff
@@ -13,8 +13,10 @@
 #ifdef __cplusplus
 extern "C" {
 #else
+#if __STDC_VERSION__ >= 201112L
 #define static_assert _Static_assert
```

Comment:
Why is this being defined at all here?, there does not seem to be any in-tree usage for this in particular(?), but we do have BUILD_ASSERT defined in include/toolchain/gcc.h

Reply:
> Why do we keep this defined at all here?

Because this header provides `<assert.h>` for toolchains that aren't using another libc, and this macro is required to be present in C11 and C18.  It may not be used in-tree, but it has been used in external systems like NRF Connect, and may be used in other systems we don't see.

Reply:
@pabigot Ok.. I guess it just depends on how minimal we see minimal libC being :)